app/tests/archives_tests/test_forms.py

```diff
@@ -201,6 +205,46 @@ def create_archive():
     assert str(archive.social_image.x20.url) in response.content.decode()
 
 
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "form_class",
+    (ArchiveItemCreateForm, ArchiveItemUpdateForm),
+)
+@pytest.mark.parametrize(
+    "existing_title,new_title,expected_validity",
+    (
+        ("", "", True),
+        ("Foo", "", True),
+        ("", "Bar", True),
+        ("Foo", "Bar", True),
+        ("Foo", "Foo", False),
+    ),
+)
+def test_archive_item_form_unique_title(
+    form_class, existing_title, new_title, expected_validity
+):
+    archive = ArchiveFactory()
+    user = UserFactory()
+    archive.add_editor(user)
+
+    ArchiveItemFactory(archive=archive, title=existing_title)
+
+    instance = None
+    if form_class == ArchiveItemUpdateForm:
+        instance = ArchiveItemFactory(archive=archive)
+
+    form = form_class(
+        user=user,
+        instance=instance,
+        base_obj=archive,
+        data={
+            "title": new_title,
+        },
+    )
+
+    assert form.is_valid() is expected_validity
```

Comment:
This test could also be refactored if you refactor the form.

Reply:
Done! Or rather, I have created a test for just the refactored Mixin